test/unit/c_api/factorization/test_cholesky_c_api.cpp

```diff
@@ -0,0 +1,243 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <complex>
+#include <functional>
+#include <sstream>
+#include <tuple>
+#include <type_traits>
+
+#include <pika/init.hpp>
+#include <pika/runtime.hpp>
+
+#include <dlaf/communication/communicator_grid.h>
+#include <dlaf/factorization/cholesky.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_mirror.h>
+#include <dlaf_c/grid.h>
+#include <dlaf_c/init.h>
+
+#include "config.h"
+#include "test_cholesky_c_api_wrapper.h"
+
+#include <gtest/gtest.h>
+
+#ifdef DLAF_WITH_SCALAPACK
+#include <dlaf_test/blacs.h>
+#endif
+#include <dlaf_test/comm_grids/grids_6_ranks.h>
+#include <dlaf_test/matrix/util_generic_lapack.h>
+#include <dlaf_test/matrix/util_matrix.h>
+#include <dlaf_test/util_types.h>
+
+using namespace dlaf;
+using namespace dlaf::comm;
+using namespace dlaf::matrix;
+using namespace dlaf::matrix::test;
+using namespace dlaf::test;
+using namespace testing;
+
+::testing::Environment* const comm_grids_env =
+    ::testing::AddGlobalTestEnvironment(new CommunicatorGrid6RanksCAPIEnvironment);
+
+template <class T>
+struct CholeskyTestMC : public TestWithCommGrids {};
+
+TYPED_TEST_SUITE(CholeskyTestMC, MatrixElementTypes);
+
+#ifdef DLAF_WITH_GPU
+template <class T>
+struct CholeskyTestGPU : public TestWithCommGrids {};
+
+TYPED_TEST_SUITE(CholeskyTestGPU, MatrixElementTypes);
+#endif
+
+const std::vector<blas::Uplo> blas_uplos({blas::Uplo::Lower, blas::Uplo::Upper});
+
+const std::vector<std::tuple<SizeType, SizeType>> sizes = {
+    //    {0, 2},  // m = 0
+    // {5, 8}, {34, 34},                    // m <= mb
+    // {4, 3}, {16, 10}, {34, 13}, {32, 5},  // m > mb
+    {4, 1},
+    //   {34, 13},
+    {32, 5}  // m > mb
```

Comment:
Why are these test-cases disabled?

Reply:
This is what I'm working on at the moment, to re-enable these tests too. The reason for now is that for such cases `mat_h.readwrite(LocalTileIndex(0, 0))` fails.